### PR TITLE
Longest road

### PR DESCRIPTION
**Pull request by delmaler**
added update_longest_road method that works everytime you build a new road.

### CODE REVIEW

File: Board.py [hunk 1]
```diff
@@ -205,30 +205,40 @@ def __init__(self, board, owner=None):
         self.temp_build_info = {}
         self.traveled = False
 
+    # need to check if owner is player or int
```
Comment (shay-halimi) [inline Board.py:208]: delete

File: Board.py [hunk 2]
```diff
@@ -205,30 +205,40 @@ def __init__(self, board, owner=None):
...
+            cur_max = road_length + 1
+            self.traveled = True
+            for cross_road in self.neighbors: #type: Crossroad
+                for neighbor in cross_road.neighbors:
+                    cur_road = neighbor.get_road()
+                    if cur_road.owner == player:
+                        cur_value = cur_road.update_longest_road(player, road_length + 1)
+                        if cur_value > cur_max:
+                            cur_max = cur_value
+            self.traveled
```
Comment (shay-halimi) [inline Board.py:223]: self.traveled = False

File: Board.py [hunk 3]
```diff
@@ -205,30 +205,40 @@ def __init__(self, board, owner=None):
...
+            return cur_value
```
Comment (shay-halimi) [inline Board.py:224]: cur_value is not defined in here

File: Board.py (hunk 1 above)
Comment (delmaler) [reply Board.py]: resolved

File: Board.py [hunk 4]
```diff
@@ -205,30 +205,39 @@ def __init__(self, board, owner=None):
         self.temp_build_info = {}
         self.traveled = False
 
+    def update_longest_road(self, player, road_length):
+        assert self.owner == player
+        if self.traveled:
+            return road_length
+        else:
```
Comment (shay-halimi) [inline Board.py:212]: testing

File: Board.py (hunk 2 above)
Comment (delmaler) [reply Board.py]: done

File: Board.py (hunk 3 above)
Comment (delmaler) [reply Board.py]: changed to max_value

**Pull request merged**